src/ObjectUtils/CoffFile.cpp

```diff
@@ -662,9 +680,71 @@ CoffFileImpl::LoadExceptionTableEntriesAsSymbols() {
     symbol_info->set_address(unwind_range.start);
     symbol_info->set_size(unwind_range.end - unwind_range.start);
   }
+
+  if (module_symbols.symbol_infos().empty()) {
+    return ErrorMessage{
+        absl::StrCat(kErrorMessagePrefix, "not even a single address range found.")};
+  }
   return module_symbols;
 }
 
+ErrorMessageOr<ModuleSymbols> CoffFileImpl::LoadDynamicLinkingSymbolsAndUnwindRangesAsSymbols() {
+  static constexpr std::string_view kErrorMessagePrefix = "Unable to load fallback symbols: ";
+
+  ErrorMessageOr<std::vector<UnwindRange>> unwind_ranges_or_error = GetUnwindRanges();
+  if (!unwind_ranges_or_error.has_value()) {
+    return ErrorMessage{
+        absl::StrCat(kErrorMessagePrefix, unwind_ranges_or_error.error().message())};
+  }
+
+  ErrorMessageOr<ModuleSymbols> dynamic_linking_symbols =
+      LoadSymbolsFromExportTableInternal(unwind_ranges_or_error.value());
+  ErrorMessageOr<ModuleSymbols> unwind_ranges_as_symbols =
+      LoadExceptionTableEntriesAsSymbolsInternal(unwind_ranges_or_error.value());
+  if (!dynamic_linking_symbols.has_value() && !unwind_ranges_as_symbols.has_value()) {
+    return ErrorMessage{absl::StrFormat("%s1) %s 2) %s", kErrorMessagePrefix,
+                                        dynamic_linking_symbols.error().message(),
+                                        unwind_ranges_as_symbols.error().message())};
+  }
+  std::vector<SymbolInfo> fallback_symbols;
+
+  absl::flat_hash_set<uint64_t> dynamic_linking_addresses;
+  if (dynamic_linking_symbols.has_value()) {
+    for (SymbolInfo& symbol_info : *dynamic_linking_symbols.value().mutable_symbol_infos()) {
+      dynamic_linking_addresses.insert(symbol_info.address());
+      // Remember that the Export Table doesn't contain the size of symbols. For leaf functions,
+      // we couldn't extract the size from unwind information, so we had set it to zero. Let's set
+      // those zeros back to our placeholder, so that later we can deduce these sizes with
+      // DeduceDebugSymbolMissingSizesAsDistanceFromNextSymbol.
+      if (symbol_info.size() == 0) {
+        symbol_info.set_size(SymbolsFile::kUnknownSymbolSize);
+      }
+      fallback_symbols.emplace_back(std::move(symbol_info));
+    }
+  }
+
+  if (unwind_ranges_as_symbols.has_value()) {
+    for (SymbolInfo& symbol_info : *unwind_ranges_as_symbols.value().mutable_symbol_infos()) {
+      if (dynamic_linking_addresses.contains(symbol_info.address())) {
```

Comment:
I might miss something, but what about the case that the dynamic linking symbol does not have a size. Can't we use the one from the  unwinding range here?

Reply:
Nope, we already assign the sizes from unwind info in `LoadSymbolsFromExportTableInternal`, so the exported functions that don't have a size yet are exactly for which we don't have unwind info. This is basically explained in the comment above. It's the leaf functions in the Microsoft definition.

Reply:
Okay. Thanks for the refresher!